File: 28.implement-str-str.js
/*
 * @lc app=leetcode id=28 lang=javascript
 *
 * [28] Implement strStr()
 */

// @lc code=start
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function(haystack, needle) {
  if (haystack === needle || needle === "") {
    return 0;
  }
  const haystackArray = haystack.split(needle);

  if (haystackArray.length <= 1) {
    return -1;
  }

  return haystackArray[0].length;
};
// @lc code=end
// 74/74 cases passed (44 ms)
// Your runtime beats 98.93 % of javascript submissions
// Your memory usage beats 89.29 % of javascript submissions (33.7 MB)

File: 28.implement-str-str.java
/*
 * @lc app=leetcode id=28 lang=java
 *
 * [28] Implement strStr()
 */

// @lc code=start
class Solution {
    public int strStr(String s, String t) {
        if (t.isEmpty())
            return 0; // edge case: "",""=>0 "a",""=>0
        for (int i = 0; i <= s.length() - t.length(); i++) {
            if (s.substring(i, i + t.length()).equals(t)) {
                return i;
            }
        }
        return -1;
    }
}
// @lc code=end
// 74/74

// cases passed (0 ms)
// Your runtime beats 100 % of java submissions
// Your memory usage beats 70.23 % of java submissions (37.6 MB)